source/uwp/Renderer/lib/AdaptiveCard.cpp

```diff
@@ -95,10 +96,27 @@ namespace AdaptiveCards { namespace Rendering { namespace Uwp
                 try
                 {
                         const double c_rendererVersion = 1.0;
-                        std::shared_ptr<::AdaptiveCards::AdaptiveCard> sharedAdaptiveCard = ::AdaptiveCards::AdaptiveCard::DeserializeFromString(jsonString, c_rendererVersion, sharedModelElementParserRegistration, sharedModelActionParserRegistration);
+                        std::shared_ptr<::AdaptiveCards::ParseResult> sharedParseResult = ::AdaptiveCards::AdaptiveCard::DeserializeFromString(jsonString, c_rendererVersion, sharedModelElementParserRegistration, sharedModelActionParserRegistration);
                         ComPtr<IAdaptiveCard> adaptiveCard;
-                        RETURN_IF_FAILED(MakeAndInitialize<AdaptiveCard>(&adaptiveCard, sharedAdaptiveCard));
+                        RETURN_IF_FAILED(MakeAndInitialize<AdaptiveCard>(&adaptiveCard, sharedParseResult->GetAdaptiveCard()));
                         RETURN_IF_FAILED(adaptiveParseResult->put_AdaptiveCard(adaptiveCard.Get()));
+
+                        for (auto sharedWarning : sharedParseResult->GetWarnings())
+                        {
+                            ComPtr<IVector<IAdaptiveWarning*>> warnings;
+                            RETURN_IF_FAILED(adaptiveParseResult->get_Warnings(&warnings));
```

Comment:
You can pull this out of the loop.

Reply:
Good point

---
In reply to: [172677757](https://github.com/Microsoft/AdaptiveCards/pull/1222#discussion_r172677757) [](ancestors = 172677757)